packages/ag-charts-community/src/chart/data/dataController.ts

```diff
@@ -97,6 +107,31 @@ export class DataController {
         }
     }
 
+    private extractScopedData(id: string, processedData: UngroupedData<any>) {
```

Comment:
If there are not multiple data-sources in use, it would be good to short-circuit this processing, to avoid adding more overhead to the 'normal' case.

(Additionally if clients are concerned about performance, we can offer them the option of using a single data-source in order to optimise our processing)